src/test/ui/async-await/async-fn-size-uninit-locals.rs

```diff
@@ -6,6 +6,7 @@
 
 // ignore-emscripten (sizes don't match)
 // run-pass
+// needs-unwind Async futures off by one byte due to panic byte flag required
```

Comment:
Ditto here.

Reply:
For this test, 3 fail on abort:
```
assert_eq!(3, std::mem::size_of_val(&single_with_noop())); # unwind
assert_eq!(2, std::mem::size_of_val(&single_with_noop())); # abort

assert_eq!(3078, std::mem::size_of_val(&joined())); # unwind
assert_eq!(3074, std::mem::size_of_val(&joined())); # abort

assert_eq!(3078, std::mem::size_of_val(&joined_with_noop())); # unwind
assert_eq!(3074, std::mem::size_of_val(&joined_with_noop())); # abort
```
I'm not seeing any patterns amongst these failure cases and the ones that pass, and am still troubleshooting. Making this comment in case anything is obvious from these changes!